### PR TITLE
Display fullname in suggestions if possible

### PR DESCRIPTION
**Pull request by cedricmessiant**
Use fullname if we have it. Else, use primary email.

### CODE REVIEW

File: src/sharing/components/ContactSuggestion.jsx [hunk 1]
```diff
@@ -32,12 +31,17 @@ export const ContactSuggestion = ({ contactOrGroup, contacts, t }) => {
...
-    const email = Contact.getPrimaryEmail(contactOrGroup)
-    name = email || getDisplayName(contactOrGroup)
-    avatarText =
-      Contact.getInitials(email) || Contact.getInitials(contactOrGroup)
+    if (contactOrGroup.fullname) {
+      name = contactOrGroup.fullname
+      avatarText = Contact.getInitials(contactOrGroup)
+    } else {
+      name = Contact.getPrimaryEmail(contactOrGroup)
+      avatarText = name[0].toUpperCase()
```
Comment (Crash--) [inline src/sharing/components/ContactSuggestion.jsx]: 'Passing a string to Contact.getInitials will be deprecated soon.' :D

Comment (cedricmessiant) [reply src/sharing/components/ContactSuggestion.jsx]: yeah, that's why I use this :-)

Comment (Crash--) [reply src/sharing/components/ContactSuggestion.jsx]: Is it possible to have a contactOrGroup without email? Should we test if name[0] exists ?

File: src/models/Contact.js [hunk 2]
```diff
@@ -0,0 +1,26 @@
+import { Contact as OriginalContact } from 'cozy-doctypes'
```
Comment (Crash--) [inline src/models/Contact.js]: DoctypeContact ?

File: src/models/Contact.js [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+  static isContact(candidate) {
+    return candidate._type === OriginalContact.doctype
+  }
+
+  static getInitials(contactOrRecipient) {
+    if (Contact.isContact(contactOrRecipient)) {
+      return OriginalContact.getInitials(contactOrRecipient)
+    } else {
+      const s = contactOrRecipient.public_name || contactOrRecipient.name
+      return s[0].toUpperCase() || ''
```
Comment (Crash--) [inline src/models/Contact.js]: Can we have `s === undefined` or ` === null` ? Should we test if s[0] exists?

File: src/models/Contact.js (hunk 2 above)
Comment (cedricmessiant) [reply src/models/Contact.js]: :+1:

**Comment by cedricmessiant** [comment]
Made the changes and added some tests. I wonder if we should move `isContact` to `cozy-doctypes`

**Comment by Crash--** [comment]
i think we should

**Comment by cedricmessiant** [comment]
waiting for https://github.com/cozy/cozy-libs/pull/418 to be able to merge

**Pull request merged**